pkg/pool/pool.go

```diff
@@ -61,10 +66,21 @@ func NewWithCapacity(workers, capacity int) Interface {
 		go func() {
 			for work := range i.workCh {
 				func() {
+					done := make(chan bool, 1)
```

Comment:
Can this be the usual `done := make(chan struct{})`, which is just closed once the func is done?